deploy/addons/kubevirt/README.md

```diff
@@ -0,0 +1,45 @@
+## kubevirt Addon
+[kubevirt](https://kubevirt.io/) - KubeVirt technology addresses the needs of development teams that have adopted or want to adopt Kubernetes but possess existing Virtual Machine-based workloads that cannot be easily containerized.
+
+See official [Minikube Quickstart](https://kubevirt.io/quickstart_minikube/) documentation for more information and to install KubeVirt without using the addon.
+
+### Enable kubevirt on minikube
+Minikube can be started with default values and those will be enough to run a quick example, that being said, if you can spare a few more GiBs of RAM (by default it uses 2GiB), it’ll allow you to experiment further.
+
+We’ll create a profile for KubeVirt so it gets its own settings without interfering what any configuration you might have already, let’s start by increasing the default memory to 4GiB:
+
+```shell script
+minikube config -p kubevirt set memory 4096
+minikube config -p kubevirt set vm-driver kvm2
+minikube start -p kubevirt
+```
+
+To enable this addon, simply run:
+```shell script
+minikube addons enable kubevirt-provisioner
+```
+
+You will then need to check if your VM’s CPU supports virtualization extensions, you can check by executing the following command:
+```shell script
+minikube ssh -p kubevirt "egrep 'svm|vmx' /proc/cpuinfo"
+```
+If the command does not generate any output, create the following ConfigMap so that KubeVirt uses emulation mode, otherwise skip to the next section:
+```shell script
+kubectl create configmap kubevirt-config -n kubevirt --from-literal debug.useEmulation=true
+```
+
+Finally enable kubevirt:
+```shell script
+minikube addons enable kubevirt
+```
+
+In a minute or so kubevirt's default components will be installed into your cluster.
+You can run `kubectl get pods -n kubevirt` to see the progress of the kubevirt installation.
+
+
+### Disable kubevirt
+To disable this addon, simply run:
+```shell script
+minikube addons disable kubevirt
+minikube addons disable kubevirt-provisioner
```

Comment:
@ashleyschuett what about sqashing these two commands into one (just kubevirt).
If somebody wants to separate operator deployment from kubevirt deployment then the user can just do  it manually using the yamls.

Reply:
After looking into this a bit and trying to get this to work I am unable to. 

The issue is if we enable kubevirt with a single command, it then also needs to be able to be disabled. When disabling all the manifests get removed at the same time. In this case, the operator deletes much quicker than the KubeVirt CR. When this happens the CR will still have a finalizer and will get stuck in a terminating state along with the namespace since the operator is gone and unable to remove it. 

I attempted to hook into the "preStop" lifecycle of the container but while the pod will then wait to delete it will still be in the "terminating" state meaning that the validating webhook will not respond and the kubevirt cr will not delete. 

Because of this, I think the options are to keep the deployments/commands as is. Or if we change them to be `kubevirt` and `kubevirt-with-emulation` then in the read me we have the user create the kubevirt CR, and let them know to remove it before disabling the addon. Does either of the options sound okay? Or is there another way to get around the previously mentioned issue?

Reply:
Nice catch. And sad.

Now, I dislike to say it, but what about writing a pod which does all this:
1. Deployment on creation and start
2. Undeploy on deletion of the pod

The reason that I think that this could work is, because we can set a terminationGrace period on the pod - which gives the pod some time to complete - and inside the pod we can just serialize the deleition of the CR and the followed deletion of the operator.

What do you think?

Reply:
I think what you mean by this is that the enable/disable would only be deploying a pod, which would then have / run a script that would match the minikube quick start documentation we have?  Then on disable delete the resources in the proper order? 

The only problem I see what that is that namespace would still need to be created first in the manifests, and it would then also be deleted first which would mean that the operator deployment could be deleted before the pod runs the deletion. 

I have changed the minikube code to delete resources in the opposite that they are created in, but i think that would need buy in from more people in this project.

Reply:
I have update the addon to use your suggestions, let me know what you think!